packages/open-payments/src/types.ts

```diff
@@ -72,3 +73,36 @@ export const isInteractiveGrant = (
 export const isNonInteractiveGrant = (
   grant: InteractiveGrant | NonInteractiveGrant
 ): grant is NonInteractiveGrant => !!(grant as NonInteractiveGrant).access_token
+
+type ASExternalComponents = ASExternal['schemas.yaml']['components']['schemas']
+export type AccessIncomingActions =
+  ASExternalComponents['access-incoming']['actions']
+export type AccessOutgoingActions =
+  ASExternalComponents['access-outgoing']['actions']
+export type AccessQuoteActions = ASExternalComponents['access-quote']['actions']
+
+export type AccessType =
+  | ASExternalComponents['access-incoming']['type']
+  | ASExternalComponents['access-outgoing']['type']
+  | ASExternalComponents['access-quote']['type']
+
+export type Action = (
+  | AccessIncomingActions
+  | AccessOutgoingActions
+  | AccessQuoteActions
+)[number]
+
+export const AccessTypeMapping: Record<string, AccessType> = Object.freeze({
+  IncomingPayment: 'incoming-payment',
+  OutgoingPayment: 'outgoing-payment',
+  Quote: 'quote'
+})
+
+export const ActionMapping: Record<string, Action> = Object.freeze({
```

Comment:
I _think_ this works and would avoid a lot of rename changes
```suggestion
export type AccessAction = (
  | AccessIncomingActions
  | AccessOutgoingActions
  | AccessQuoteActions
)[number]

export const AccessType: Record<string, AccessType> = Object.freeze({
  IncomingPayment: 'incoming-payment',
  OutgoingPayment: 'outgoing-payment',
  Quote: 'quote'
})

export const AccessAction: Record<string, AccessAction> = Object.freeze({
```
(I vote for backend's `AccessAction` over auth's `Action`, but I'm biased)

Reply:
the naming is better this way, thank you for the suggestion